### ISSUE TITLE
关于FAN替换MLP在时序预测中实验的问题

### ISSUE THREAD

**Issue by Halthward** [YihongDong/FAN#3]
注意到您尝试对时序预测中transformer的ffn部分mlp的结构做替换 在论文中提到使用了三种设置来完成实验
class EncoderLayer(nn.Module):
    def __init__(self, attention, d_model, d_ff=None, dropout=0.1, activation="relu", exp_setting=0):
        super(EncoderLayer, self).__init__()
        self.exp_setting = exp_setting
        d_ff = d_ff or 4 * d_model
        self.attention = attention
        assert exp_setting in [0, 2, 4]
        if exp_setting == 0:
            self.mlp1 = nn.Linear(d_model, d_ff)
            self.mlp2 = nn.Linear(d_ff, d_model)
        elif exp_setting == 2:
            self.mlp1 = FANLayer(input_dim=d_model, output_dim=d_ff, with_gate=True)
            self.mlp2 = FANLayer(input_dim=d_ff, output_dim=d_model, with_gate=True)
        elif exp_setting == 4:
            self.mlp1 = FANLayer(input_dim=d_model, output_dim=d_ff, with_gate=False)
            self.mlp2 = FANLayer(input_dim=d_ff, output_dim=d_model, with_gate=False)
        
        self.norm1 = nn.LayerNorm(d_model)
        self.norm2 = nn.LayerNorm(d_model)
        self.dropout = nn.Dropout(dropout)
        self.activation = F.relu if activation == "relu" else F.gelu

    def forward(self, x, attn_mask=None):
        new_x, attn = self.attention(
            x, x, x,
            attn_mask=attn_mask
        )
        x = x + self.dropout(new_x)

        y = x = self.norm1(x)
        y = self.dropout(self.activation(self.mlp1(y)))
        y = self.dropout(self.mlp2(y))

        return self.norm2(x + y), attn
但是在您论文之中只有性能对比的表格 我想问的问题是如何解释对于同一结构 不同的预测时间步长下FAN替换MLP发挥作用的不同 有的时候甚至是副作用呢 有没有一个统一的可解释性的规律呢 您是否做了相关的可解释性支撑的实验
<img width="527" alt="微信图片_20241129163944" src="https://github.com/user-attachments/assets/4c15f594-6a53-4fcc-8c06-ad6eedf1075c">

**Comment by YihongDong** [comment]
您好，很高兴您能关注我们的工作，在此解释一下：附录里的Table 5这个实验是因为有人联系我说FAN在他们项目里的效果比较不错，并建议我们在时序预测任务上使用 instance normalization来提升所有模型的整体效果，然后我们进行了另外的实验并如实report了结果（原实验可见Table 2），暂时还没有进行额外的可解释性分析。据我们的实验观察，各种normalization对周期性建模是存在一定影响的，可能需要进行针对性的修改。我们也欢迎您进行深入的分析。
